### PR TITLE
Implement diamond - port from ECMAScript

### PR DESCRIPTION
**Pull request by njgingrich**
Implements #102, referencing #37.

### CODE REVIEW

**Comment by masters3d** [comment]
"-> An implementation for 'diamond' was found, but config.json does not reference this exercise."

File: exercises/diamond/diamond.test.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+describe('Make diamond function', () => {
+  const diamond = new Diamond()
+
+  test('test letter A', () => {
+    const result = 'A\n'
+    expect(diamond.makeDiamond('A')).toEqual(result)
+  })
+
+  test('test letter C', () => {
```
Comment (masters3d) [inline exercises/diamond/diamond.test.ts]: This can be a multiline string?

**Comment by njgingrich** [comment]
I added the diamond section from `config.json` in the ecmascript branch - If that's incorrect, let me know what I should to do make it proper and I will.

**Comment by masters3d** [comment]
@njgingrich  I updated the some of the files which makes one of tests fail on your branch. Can you rebase you branch and copy the common files in your exercise?

**Comment by masters3d** [comment]
> I added the diamond section from config.json in the ecmascript branch - If that's incorrect, let me know what I should to do make it proper and I will.

@njgingrich  I did not see a entry added in the config.json in this commit 
1d88ff2

I went ahead an rebase your branch and updated the common files. 

Do you want to add the config.json entry? Then I can merge this.

**Comment by njgingrich** [comment]
Hi, sorry about the delay - I will add the config entry.

**Comment by njgingrich** [comment]
Sorry I think I may just be missing something - I can't decipher the failing checks to see what I need to do.

**Comment by masters3d** [comment]
Try to Remove this 
5f03f9a merge from remote

That merge undid some of the previous fixes in your branch.

**Pull request merged**